modin/pandas/dataframe.py

```diff
@@ -2496,18 +2496,22 @@ def __setattr__(self, key, value):
         # - `_siblings`, which Modin initializes before it appears in __dict__
         if key in ["_query_compiler", "_siblings"] or key in self.__dict__:
             pass
-        elif key in self and key not in dir(self):
+        elif key not in dir(self) and key not in self and not is_list_like(value):
+            # Assigning a scalar to a non-existing attribute should not result in a
+            # broadcasted column, but rather just a new attribute.
+            pass
+        elif key not in dir(self):
+            if key not in self:
+                warnings.warn(
+                    "Modin doesn't allow columns to be created via a new attribute name - see "
+                    + "https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access",
+                    UserWarning,
+                )
             self.__setitem__(key, value)
             # Note: return immediately so we don't keep this `key` as dataframe state.
             # `__getattr__` will return the columns not present in `dir(self)`, so we do not need
             # to manually track this state in the `dir`.
             return
-        elif isinstance(value, pandas.Series):
```

Comment:
We could add an extra `elif` case that checks to see if the value the user is trying to set `is_list_like()` here and throw the warning in this case.

Reply:
I think replacement is sufficient rather than having both checks, right?

Reply:
Replacement is sufficient since `is_list_like()` will always be True for `pandas.Series`.